src/sinks/aws_sqs.rs

```diff
@@ -0,0 +1,412 @@
+}
+
+lazy_static! {
+    static ref REQUEST_DEFAULTS: TowerRequestConfig = TowerRequestConfig {
+        timeout_secs: Some(30),
+        ..Default::default()
+    };
+}
+
+#[derive(Deserialize, Serialize, Debug, Eq, PartialEq, Clone, Derivative)]
+#[serde(rename_all = "snake_case")]
+pub enum Encoding {
+    Text,
+    Json,
+}
+
+inventory::submit! {
+    SinkDescription::new::<SqsSinkConfig>("aws_sqs")
+}
+
+impl GenerateConfig for SqsSinkConfig {
+    fn generate_config() -> toml::Value {
+        toml::from_str(
+            r#"queue_url = "https://sqs.us-east-2.amazonaws.com/123456789012/MyQueue"
+            region = "us-east-2"
+            encoding.codec = "json""#,
+        )
+        .unwrap()
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "aws_sqs")]
+impl SinkConfig for SqsSinkConfig {
+    async fn build(
+        &self,
+        cx: SinkContext,
+    ) -> crate::Result<(super::VectorSink, super::Healthcheck)> {
+        let client = self.create_client()?;
+        let healthcheck = self.clone().healthcheck(client.clone());
+        let sink = SqsSink::new(self.clone(), cx, client)?;
+        Ok((super::VectorSink::Sink(Box::new(sink)), healthcheck.boxed()))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn sink_type(&self) -> &'static str {
+        "aws_sqs"
+    }
+}
+
+impl SqsSinkConfig {
+    pub async fn healthcheck(self, client: SqsClient) -> crate::Result<()> {
+        client
+            .get_queue_attributes(GetQueueAttributesRequest {
+                attribute_names: None,
+                queue_url: self.queue_url.clone(),
+            })
+            .await
+            .map(|_| ())
+            .context(GetQueueAttributes)
+            .map_err(Into::into)
+    }
+
+    pub fn create_client(&self) -> crate::Result<SqsClient> {
+        let region = (&self.region).try_into()?;
+        let client = rusoto::client()?;
+
+        let creds = rusoto::AwsCredentialsProvider::new(&region, self.assume_role.clone())?;
+
+        Ok(SqsClient::new_with(client, creds, region))
+    }
+}
+
+impl SqsSink {
+    pub fn new(
+        config: SqsSinkConfig,
+        cx: SinkContext,
+        client: SqsClient,
+    ) -> crate::Result<impl Sink<Event, Error = ()>> {
+        // Currently we do not use batching, so this mostly for future. Also implement `Service` is simpler than `Sink`.
+        // https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-batch-api-actions.html
+        // Up to 10 events, not more than 256KB as total size.
+        let batch = BatchSettings::default().events(1).bytes(262_144);
+
+        let request = config.request.unwrap_with(&REQUEST_DEFAULTS);
+        let encoding = config.encoding;
+        let fifo = config.queue_url.ends_with(".fifo");
+        let message_group_id = match config.message_group_id {
+            Some(value) => Some(Template::try_from(value).context(TopicTemplate)?),
```

Comment:
I think we should add an extra `Some(value) if fifo` that returns an error as message_group_id cannot be used with a non-fifo queue.